src/model/entities/rundown.ts

```diff
@@ -350,6 +351,23 @@ export class Rundown extends BasicRundown {
     segment?.removeUnsyncedParts()
   }
 
+  private setSegmentExecutedAtEpochTime(executedAtEpochTime: number): void {
+    const onAirSegment: Segment | undefined = this.activeCursor?.segment
+    const nextSegment: Segment | undefined = this.nextCursor?.segment
+
+    if (onAirSegment?.id === nextSegment?.id) {
+      return
+    }
+
+    if (onAirSegment) {
+      onAirSegment.clearExecutedAtEpochTime()
+    }
+
+    if (nextSegment) {
+      nextSegment.setExecutedAtEpochTime(executedAtEpochTime)
+    }
+  }
```

Comment:
This seems like something that should be handled by the Segment itself.
When the Segment is put onAir it should set its executedAtTime. When it's taking ofAir, it should clear it. No need to let the Rundown handled that logic.

Reply:
You are absolutely right. It is updated :)